rasa/core/events/__init__.py

```diff
@@ -632,87 +651,143 @@ def __eq__(self, other) -> bool:
         else:
             return self.name == other.name
 
-    def __str__(self) -> str:
+    def __str__(self) -> Text:
         return (
-            "ReminderScheduled("
-            "action: {}, trigger_date: {}, name: {}"
-            ")".format(self.action_name, self.trigger_date_time, self.name)
+            f"ReminderScheduled(intent: {self.intent}, trigger_date: {self.trigger_date_time}, "
+            f"entities: {self.entities}, name: {self.name})"
         )
 
-    def _data_obj(self) -> Dict[Text, Any]:
+    def scheduled_job_name(self, sender_id: Text) -> Text:
+        return (
+            f"[{hash(self.name)},{hash(self.intent)},{hash(str(self.entities))}]"
+            + ACTION_NAME_SENDER_ID_CONNECTOR_STR
+            + sender_id
+        )
+
+    def _properties(self) -> Dict[Text, Any]:
         return {
-            "action": self.action_name,
+            "intent": self.intent,
             "date_time": self.trigger_date_time.isoformat(),
+            "entities": self.entities,
             "name": self.name,
             "kill_on_user_msg": self.kill_on_user_message,
         }
 
     def as_story_string(self) -> Text:
-        props = json.dumps(self._data_obj())
+        props = json.dumps(self._properties())
         return f"{self.type_name}{props}"
 
     def as_dict(self) -> Dict[Text, Any]:
         d = super().as_dict()
-        d.update(self._data_obj())
+        d.update(self._properties())
         return d
 
     @classmethod
     def _from_story_string(cls, parameters: Dict[Text, Any]) -> Optional[List[Event]]:
 
         trigger_date_time = parser.parse(parameters.get("date_time"))
+
         return [
             ReminderScheduled(
-                parameters.get("action"),
+                parameters.get("intent"),
                 trigger_date_time,
-                parameters.get("name", None),
-                parameters.get("kill_on_user_msg", True),
-                parameters.get("timestamp"),
-                parameters.get("metadata"),
+                parameters.get("entities"),
+                name=parameters.get("name"),
+                kill_on_user_message=parameters.get("kill_on_user_msg", True),
+                timestamp=parameters.get("timestamp"),
+                metadata=parameters.get("metadata"),
             )
         ]
 
 
 # noinspection PyProtectedMember
 class ReminderCancelled(Event):
-    """Cancel all jobs with a specific name."""
+    """Cancel certain jobs."""
 
     type_name = "cancel_reminder"
 
     def __init__(
         self,
-        action_name: Text,
+        name: Optional[Text] = None,
+        intent: Optional[Text] = None,
+        entities: Optional[List[Dict]] = None,
         timestamp: Optional[float] = None,
         metadata: Optional[Dict[Text, Any]] = None,
     ):
-        """
+        """Creates a ReminderCancelled event.
+
+        All arguments are optional and used to refine the set of reminders that
```

Comment:
optional: I'd shorten it to
```suggestion
        If all arguments are `None`, this will cancel all reminders.
```